openassessment/data.py

```diff
@@ -1577,3 +1579,122 @@ def get_file_uploads(self, missing_blank=False):
+    return scorer_name, scorer_username, scorer_email
+
+def generate_assessment_data(assessment_list):
+    results = []
+    for assessment in assessment_list:
+
+        scorer_name, scorer_username, scorer_email = get_scorer_data(assessment.scorer_id)
+
+        assessment_data = {
+            "idAssessment": str(assessment.id),
+            "grader_name": scorer_name,
+            "grader_username": scorer_username,
+            "grader_email": scorer_email,
+            "assesmentDate": assessment.scored_at.strftime('%d-%m-%Y'),
+            "assesmentScores": parts_summary(assessment),
+            "problemStep": score_type_to_string(assessment.score_type),
+            "feedback": assessment.feedback or ''
+        }
+
+        results.append(assessment_data)
+    return results
+
+def generate_received_assessment_data(submission_uuid=None):
+    """
+    Generates a list of received assessments data based on the submission UUID.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+
+    results = []
+
+    submission = None
+    if submission_uuid:
+        submission = sub_api.get_submission_and_student(submission_uuid)
+
+    if not submission:
+        return results
+
+    assessments = _use_read_replica(
+        Assessment.objects.prefetch_related('parts').
+        prefetch_related('rubric').
+        filter(
+            submission_uuid=submission['uuid']
+        )
+    )
+    return generate_assessment_data(assessments)
+
+
+def generate_given_assessment_data(item_id=None, submission_uuid=None):
+    """
+    Generates a list of given assessments data based on the submission UUID as scorer.
+
+    Args:
+        submission_uuid (str, optional): The UUID of the submission. Defaults to None.
+
+    Returns:
+        list[dict]: A list containing assessment data dictionaries.
+    """
+    results = []
+    # Getting the scorer student id
+    primary_submission = sub_api.get_submission_and_student(submission_uuid)
+
+    if not primary_submission:
+        return results
+
+    scorer_student_id = primary_submission['student_item']['student_id']
+    submissions = None
+    if item_id:
+        submissions = Submission.objects.filter(student_item__item_id=item_id).values('uuid')
+        submission_uuids = [sub['uuid'] for sub in submissions]
+
+    if not submission_uuids or not submissions:
+        return results
+
+    # Now fetch all assessments made by this student for these submissions
+    assessments_made_by_student = _use_read_replica(
+        Assessment.objects.prefetch_related('parts')
+        .prefetch_related('rubric')
+        .filter(scorer_id=scorer_student_id, submission_uuid__in=submission_uuids)
+    )
+
+    return generate_assessment_data(assessments_made_by_student)
```

Comment:
The overall solution looks good. I'll leave some suggestions for style, but overall, good!